form_manual_allocation.php

```diff
@@ -85,6 +86,21 @@ protected function render_filter() {
                 null, array(0, 1));
         $mform->setType('show_alloc_necessary', PARAM_BOOL);
 
+        // Filter by group.
+        $choicegroups = $this->ratingallocate->get_all_groups_of_choices();
+        $allgroups = array();
+        foreach ($choicegroups as $choicegroup) {
+            $allgroups[$choicegroup] = groups_get_group($choicegroup);
+        }
+        $groupsmenu[0] = get_string('allparticipants');
+        $groupsmenu[-1] = get_string('nogroup', 'enrol');
+        foreach ($allgroups as $gid => $unused) {
+            $groupsmenu[$gid] = $allgroups[$gid]->name;
+        }
+        if (count($groupsmenu) > 1) {
+            $mform->addElement('select', 'filtergroup', get_string('group'), $groupsmenu);
```

Comment:
I would suggest adding a help button for this element to explain that you can only filter for those groups which are used as an restriction for the choices.